node/tcp/src/tcp.rs

```diff
@@ -283,13 +295,6 @@ impl Tcp {
                 task.abort();
             }
 
-            // If the (owning) Tcp was not the initiator of the connection, it doesn't know the listening address
-            // of the associated peer, so the related stats are unreliable; the next connection initiated by the
-            // peer could be bound to an entirely different port number
-            if conn.side() == ConnectionSide::Initiator {
```

Comment:
@ljedrz can you confirm safety?

Reply:
this is safe; until now, the known (low-level) peers were distinguished by IP+port, and since inbound connections are assigned generic ports, this wasn't suitable for long-term peer tracking; however, moving to IPs only for that collection means that there is no need for this special-casing anymore